### PR TITLE
Closes #247. Add debug to SCollection.

### LINKED ISSUE #247: Add `debug` transformation
Add `debug` transformation to dump content of the SCollection. Useful for local debugging of tests or logic.

Technically one could use `materialize` or `textFile` etc, but in both cases it requires changes to the pipeline logic, and are not as simple as a drop in `debug` transformation known from for example scalding.

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

**Comment by nevillelyh** [comment]
This will only work with in process runner right?

**Comment by ravwojdyla** [comment]
@nevillelyh this will work on any runner. On Cloud Dataflow runners, `Console.out` goes to worker logs as `INFO`, `Console.err` goes to worker logs as `ERROR`. But again I see this as local debugging, if you plan to log anything in Cloud Dataflow, a proper Logger (+ sampling) is a better idea.

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala [hunk 1]
```diff
@@ -511,6 +511,19 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
       .toSCollection
   }
 
+  /**
+   * Print content of a SCollection to `out()`.
+   * @group debug
+   */
+  def debug(out:() => PrintStream = () => Console.out, prefix: String = ""): SCollection[T] =
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: Space after :

**Comment by nevillelyh** [comment]
:+1:

**Pull request merged**